packages/mds-geography-author/api.ts

```diff
@@ -0,0 +1,206 @@
+        await log.error('failed to edit geography', err.stack)
+        if (err instanceof NotFoundError) {
+          return res.status(404).send({ result: 'not found' })
+        }
+        return res.status(500).send(new ServerError(err))
+      }
+    }
+  )
+
+  app.delete(
+    pathsFor('/geographies/:geography_id'),
+    checkAccess(scopes => scopes.includes('policies:delete')),
+    async (req, res) => {
+      const { geography_id } = req.params
+      try {
+        await db.deleteGeography(geography_id)
+        return res.status(200).send({ result: `Successfully deleted geography of id ${geography_id}` })
+      } catch (err) {
+        await log.error('failed to delete geography', err.stack)
+        if (err instanceof NotFoundError) {
+          return res.status(404).send({ result: 'geography either not found or already published' })
+        }
+        return res.status(500).send(new ServerError())
+      }
+    }
+  )
+
+  app.get(
+    pathsFor('/geographies/:geography_id/meta'),
+    checkAccess(scopes => scopes.includes('policies:read')),
+    async (req, res) => {
+      const { geography_id } = req.params
+      try {
+        const geography_metadata = await db.readSingleGeographyMetadata(geography_id)
+        return res.status(200).send(geography_metadata)
+      } catch (err) {
+        await log.error('failed to read geography metadata', err.stack)
+        if (err instanceof NotFoundError) {
+          return res.status(404).send({ result: 'not found' })
+        }
+        return res.status(500).send(new ServerError())
+      }
+    }
+  )
+
+  app.get(
+    pathsFor('/geographies'),
+    checkAccess(scopes => scopes.includes('policies:read')),
+    async (req, res) => {
+      const summary = req.query.summary === 'true'
+      try {
+        const geographies = summary ? await db.readGeographySummaries() : await db.readGeographies()
+        return res.status(200).send(geographies)
+      } catch (err) {
+        // We don't throw a NotFoundError if the number of results is zero, so the error handling
+        // doesn't need to consider it here.
+        await log.error('failed to read geographies', err.stack)
+        return res.status(500).send({ result: 'geographies are unavailable' })
+      }
+    }
+  )
+
+  app.put(
+    pathsFor('/geographies/:geography_id/meta'),
+    checkAccess(scopes => scopes.includes('policies:write')),
+    async (req, res) => {
+      const geography_metadata = req.body
+      const { geography_id } = req.params
+      try {
+        await db.updateGeographyMetadata(geography_metadata)
+        return res.status(200).send(geography_metadata)
+      } catch (updateErr) {
+        if (updateErr instanceof NotFoundError) {
+          try {
+            await db.writeGeographyMetadata(geography_metadata)
+            return res.status(201).send(geography_metadata)
+          } catch (writeErr) {
+            await log.error('failed to write geography metadata', writeErr.stack)
```

Comment:
I think it's ok to have some logging in both layers. If the client is seeing an error because the DB is inaccessible, that should be logged, but it wouldn't get logged at the DB layer.